adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Flows/AcquireTokenNonInteractiveHandler.cs

```diff
@@ -126,8 +126,31 @@ public AcquireTokenNonInteractiveHandler(RequestData requestData, UserAssertion
                     {
                         throw new AdalException(AdalError.MissingFederationMetadataUrl);
                     }
-                    
-                    WsTrustAddress wsTrustAddress = await MexParser.FetchWsTrustAddressFromMexAsync(userRealmResponse.FederationMetadataUrl, this.userCredential.UserAuthType, RequestContext).ConfigureAwait(false);
+
+                    WsTrustAddress wsTrustAddress = null;
+                    try
+                    {
+                        var _wsTrustAddress = await Microsoft.Identity.Core.WsTrust.MexParser.FetchWsTrustAddressFromMexAsync(userRealmResponse.FederationMetadataUrl, this.userCredential.UserAuthType, RequestContext).ConfigureAwait(false);
```

Comment:
Style: prefer the using statements to fully qualified names directly in the code, because the latter reduce the ability to read the code.